### PR TITLE
Logic changed to take into account icon color in theme as well as a test

### PR DESCRIPTION
**Pull request by britt6612**
What does this PR do?

Changes the logic for the icon in a button to take the color in the theme if one is provided.
Also made a example for storybook using a custom theme and passing in a color to the icon
to provide an example
Where should the reviewer start?

src/js/components/Button/button.js
What testing has been done on this PR?

none
How should this be manually tested?
Any background context you want to provide?
What are the relevant issues?

This is related to issue #2820
Screenshots (if appropriate)
Do the grommet docs need to be updated?

auto
Should this PR be mentioned in the release notes?

no
Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
Thanks @britt6612 ! I think you need to merge master to your branch, remove node_modules and run `yarn test-updates` for new snapshots. it looks like the snapshots need to be synced with master. the tests are failing on this PR due to the snapshots.

**Comment by britt6612** [comment]
@ShimiSun Thank you I knew it had to be something with the tests snap shots I was just telling @oorestisime let me try what you said!

**Comment by oorestisime** [comment]
Fixed the tests although this needs some investigation before merging. i haven't looked yet to find why a bunch of other unrelated snapshots need updating.

**Comment by britt6612** [comment]
@oorestisime thank you!

**Comment by ShimiSun** [comment]
@britt6612 looks like the tests passed 🎉 @oorestisime are we good to go with this one from your end?

**Comment by oorestisime** [comment]
No i didn't yet check why we need to update the snapshots not related to button. (maybe they are valid but we should verify before.)

**Comment by ShimiSun** [comment]
@oorestisime I think the components with changed snapshots are using button behind the scenes and may be impacted by the change.

File: src/js/components/Button/Button.js [hunk 1]
```diff
@@ -86,9 +86,10 @@ class Button extends Component {
 
     let buttonIcon = icon;
     // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    if (icon && !icon.props.color) {
```
Comment (ericsoderberghp) [inline src/js/components/Button/Button.js]: This appears to break things: https://deploy-preview-2891--sad-tereshkova-173d07.netlify.com/?selectedKind=Button&selectedStory=Icon%20Label&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel

**Comment by britt6612** [comment]
Opened in a new branch!

**Pull request closed without merging**